js/src/behaviors/dismiss.ts

```diff
@@ -0,0 +1,42 @@
+export function initDismiss() {
+	window.addEventListener('click', ({ target }: Event) => {
+		const dismissButton = target instanceof Element && target.closest('[data-dismiss]');
+
+		if (!dismissButton) {
+			return;
+		}
+
+		const dismissTarget = dismissButton.closest('[data-dismissable]');
+
+		if (!dismissTarget) {
+			return;
+		}
+
+		dismissElement(dismissTarget);
+	});
+}
+
+async function dismissElement(dismissTarget: Element) {
```

Comment:
I'd like to propose an alternative approach that lets us keep the `prefers-reduced-motion` logic in the CSS, avoid any delays in removing the element when the users prefers reduced motion, and lets us flex to any arbitrary animation timing: the [`animationend` event](https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event).

This would need two things:
1. In the CSS, use query `prefers-reduced-motion: reduce` media to set `animation-duration: 0s`.
2. Refactor this function to look like this:

```ts
async function dismissElement(dismissTarget: Element) {
	dismissTarget.addEventListener('animationend', () => {
		dismissTarget.remove();
	});

	const dismissAnimation = dismissTarget.getAttribute('data-dismiss-animation');

	if (dismissAnimation) {
		dismissTarget.classList.add('is-dismissed');
		switch (dismissAnimation) {
			case 'slide-up':
				dismissTarget.classList.add('animation-slide-up');
				break;
			case 'fade':
				dismissTarget.classList.add('animation-fade');
				break;
		}
	}

	// Notification about content update occurrence.
	window.dispatchEvent(new CustomEvent('dismiss-content-update'));
}
```

From my local testing, this should work.

Reply:
(Alternatively, the `addEventListener` call could/should probably go inside the `if (dismissAnimation)` block.

Reply:
An `await new Promise(resolve => setTimeout(resolve, 100)` could go inside the event listener to add extra buffer onto the end of the animation duration before the element is removed.

Reply:
Thanks for sharing your idea, I slightly modified and applied it :)